packages/consensus/source/committed-block-state.ts

```diff
@@ -0,0 +1,64 @@
+import { inject, injectable, tagged } from "@mainsail/container";
+import { Contracts, Identifiers } from "@mainsail/contracts";
+
+@injectable()
+export class CommittedBlockState implements Contracts.BlockProcessor.IProcessableUnit {
+	@inject(Identifiers.WalletRepository)
+	@tagged("state", "clone")
+	private readonly walletRepository!: Contracts.State.WalletRepositoryClone;
+
+	@inject(Identifiers.ValidatorSet)
+	private readonly validatorSet!: Contracts.ValidatorSet.IValidatorSet;
+
+	#committedBlock!: Contracts.Crypto.ICommittedBlock;
+	#processorResult?: boolean;
+	#validators = new Map<string, Contracts.State.Wallet>();
+
+	get height(): number {
+		return this.#committedBlock.commit.height;
+	}
+
+	get round(): number {
+		return this.#committedBlock.commit.round;
+	}
+
+	get validators(): string[] {
+		return [...this.#validators.keys()];
+	}
+
+	public async configure(committedBlock: Contracts.Crypto.ICommittedBlock): Promise<CommittedBlockState> {
+		this.#committedBlock = committedBlock;
+
+		const validators = await this.validatorSet.getActiveValidators();
+		for (const validator of validators) {
+			const consensusPublicKey = validator.getAttribute<string>("validator.consensusPublicKey");
+			this.#validators.set(consensusPublicKey, validator);
+		}
+
+		return this;
+	}
+
+	public getWalletRepository(): Contracts.State.WalletRepositoryClone {
+		return this.walletRepository;
+	}
+
+	public getBlock(): Contracts.Crypto.IBlock {
+		return this.#committedBlock.block;
+	}
+
+	public setProcessorResult(processorResult: boolean): void {
+		this.#processorResult = processorResult;
+	}
+
+	public getProcessorResult(): boolean {
+		return !!this.#processorResult;
+	}
+
+	public setCommittedBlock(committedBlock: Contracts.Crypto.ICommittedBlock): void {
```

Comment:
Do we need the setter? (looks like it's unused)

Technically, the validator set is tied to the block height (because of milestones, vote weight etc.) but the current interface isn't establishing any coupling between `validatorSet` and `committedBlock`.

Though I don't think it will make a practical difference, since we can assume both are always in sync here. But I guess something along the lines of passing height to `getActiveValidators()` would be more clear / explicit.

Reply:
I will remove the setter. We are setting block in config. 

That is correct. We don't check the validators yet. It is a work in the progress and aggregated signature should be checked against `validatorSet` by height. 

We will add additional check to ensure that we are creating `RoundStates` that matches the `lastBlockHeight`.